vector/src/debug/java/im/vector/app/features/debug/di/FeaturesModule.kt

```diff
@@ -44,5 +50,15 @@ interface FeaturesModule {
         fun providesDebugVectorFeatures(context: Context, defaultVectorFeatures: DefaultVectorFeatures): DebugVectorFeatures {
             return DebugVectorFeatures(context, defaultVectorFeatures)
         }
+
+        @Provides
+        fun providesDefaultVectorOverrides(): DefaultVectorOverrides {
+            return DefaultVectorOverrides()
+        }
```

Comment:
It's maybe because of `@InstallIn(SingletonComponent::class)` anotation of the interface `FeaturesModule` but I am not a big specialist of the DI.

Reply:
there's two parts to this

- the default implementations avoid being directly injectable as we want to avoid leaking their implementations and ensure the consumption is only done via `VectorFeatures`
- we have separate debug and feature DI modules, debug configurations need to know the implementation details in order to provide fallbacks

```kotlin
// Both debug and release need to provide the interface VectorFeatures

// Release only provides the down scoped interface 
@Provides
fun providesDefaultVectorFeatures(): VectorFeatures {
    return DefaultVectorFeatures()
}

// Debug 
// Defaults are provided but not accessible outside debug variant, provides binding source for @Binds
@Provides
fun providesDefaultVectorFeatures(): DefaultVectorFeatures {
    return DefaultVectorFeatures()
}

// debug features which make use of the of the default features
@Provides
fun providesDebugVectorFeatures(context: Context, defaultVectorFeatures: DefaultVectorFeatures): DebugVectorFeatures {
    return DebugVectorFeatures(context, defaultVectorFeatures)
}

// Provides the debug instance as the VectorFeatures for the rest of the app
@Binds
fun bindFeatures(debugFeatures: DebugVectorFeatures): VectorFeatures
```

Reply:
I see, this does shine some light on it, thanks!